lib/modules/datasource/terraform-module/index.ts

```diff
@@ -80,8 +123,43 @@ export class TerraformModuleDatasource extends TerraformDatasource {
     if (latestVersion) {
       latestVersion.releaseTimestamp = res.published_at;
     }
+    return dep;
+  }
 
-    logger.trace({ dep }, 'dep');
+  /**
+   * this version uses the Module Registry Protocol that all registries are required to implement
+   * https://www.terraform.io/internals/module-registry-protocol
+   */
+  private async queryRegistryVersions(
+    serviceDiscovery: ServiceDiscoveryResult,
+    registry: string,
+    repository: string
+  ): Promise<ReleaseResult> {
+    let res: TerraformModuleVersions;
+    let pkgUrl: string;
+    try {
+      pkgUrl = `${registry}${serviceDiscovery['modules.v1']}${repository}/versions`;
+      res = (await this.http.getJson<TerraformModuleVersions>(pkgUrl)).body;
+      if (res.modules.length < 1) {
+        logger.warn({ pkgUrl }, 'Terraform registry result mismatch');
+        throw new Error('Terraform registry result mismatch');
```

Comment:
i think we should return null instead of throwing a error which is caught right after.